src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs

```diff
@@ -77,8 +77,8 @@ public static class WebScriptHostBuilderExtension
                     services.AddSingleton<DefaultScriptWebHookProvider>();
                     services.TryAddSingleton<IScriptWebHookProvider>(p => p.GetService<DefaultScriptWebHookProvider>());
                     services.TryAddSingleton<IWebHookProvider>(p => p.GetService<DefaultScriptWebHookProvider>());
-                    services.TryAddSingleton<IJobHostMiddlewarePipeline, DefaultMiddlewarePipeline>();
-                    services.TryAddSingleton<IJobHostHttpMiddleware, CustomHttpHeadersMiddleware>();
+                    services.TryAddEnumerable(ServiceDescriptor.Singleton<IJobHostMiddlewarePipeline, DefaultMiddlewarePipeline>());
+                    services.TryAddEnumerable(ServiceDescriptor.Singleton<IJobHostHttpMiddleware, CustomHttpHeadersMiddleware>());
```

Comment:
You can just do `AddSingleton()` as well, which will register both of them. I'm guessing the second one wasn't being registered correctly?

Reply:
Although, `TryAddEnumerable()` is a little safer, it looks like. It prevents two of the same ServiceDescriptors from being registered. So you can ignore my previous comment :-)